conn2res/plotting.py

```diff
@@ -0,0 +1,248 @@
+        if normalize == True:
+            for i in range(self.n_pcs):
+                pc[:, i] /= self.s[self.idx_pcs[i]]
+
+        return pc
+
+    def fit_transform(self, data, **kwargs):
+        # fit PCA
+        self.fit(data, **kwargs)
+
+        # transform data into principal components
+        return self.transform(data, **kwargs)
+
+
+def plot_task(x, y, title, num=1, figsize=(12, 10), savefig=False, block=True):
+
+    fig = plt.figure(num=num, figsize=figsize)
+    ax = plt.subplot(111)
+
+    # xlabels, ylabels
+    try:
+        x_labels = [f'I{n+1}' for n in range(x.shape[1])]
+    except:
+        x_labels = 'I1'
+    try:
+        y_labels = [f'O{n+1}' for n in range(y.shape[1])]
+    except:
+        y_labels = 'O1'
+
+    plt.plot(x[:], label=x_labels)
+    plt.plot(y[:], label=y_labels)
+    plt.legend()
+    plt.suptitle(title)
+
+    sns.despine(offset=10, trim=True)
+    if savefig:
+        fig.savefig(fname=os.path.join(FIG_DIR, f'{title}_io.png'),
+                    transparent=True, bbox_inches='tight', dpi=300)
+    plt.show(block=block)
+
+
+def plot_performance_curve(df, title, num=2, figsize=(12, 10), savefig=False, block=True):
+
+    sns.set(style="ticks", font_scale=2.0)
+    fig = plt.figure(num=num, figsize=figsize)
+    ax = plt.subplot(111)
+
+    palette, hue, hue_order = None, None, None
+    if 'module' in df.columns:
+        hue = 'module'
+        n_modules = len(np.unique(df['module']))
+        palette = sns.color_palette('husl', n_modules+1)[:n_modules]
+
+        if 'VIS' in list(np.unique(df['module'])):
+            hue_order = ['VIS', 'SM', 'DA', 'VA', 'LIM', 'FP', 'DMN']
+        else:
+            hue_order = None
+
+    sns.lineplot(data=df, x='alpha', y='score',
+                 hue=hue,
+                 hue_order=hue_order,
+                 palette=palette,
+                 markers=True,
+                 ax=ax)
+
+    sns.despine(offset=10, trim=True)
+    plt.title(title)
+    if savefig:
+        fig.savefig(fname=os.path.join(FIG_DIR, f'{title}_score.png'),
+                    transparent=True, bbox_inches='tight', dpi=300)
+    plt.show(block=block)
+
+
+def plot_time_series(x, feature_set='orig', idx_features=None, n_features=None, xlim=[0, 150], ylim=None,
+                     scaler=1, num=1, figsize=(12, 6), subplot=None, title=None, fname='time_course',
+                     legend_label='', savefig=False, block=True, **kwargs):
+    # transform data
+    x = transform_data(
+        x, feature_set, idx_features=idx_features, n_features=n_features, scaler=scaler, **kwargs)
+
+    # open figure and create subplot
+    plt.figure(num=num, figsize=figsize)
+    if subplot is None:
+        subplot = (1, 1, 1)
+    plt.subplot(*subplot)
+
+    # plot data
+    if x.size > 0:
+        plt.plot(x)
+
+        # add x and y limits
+        plt.xlim(xlim)
+        if ylim is not None:
+            plt.xlim(ylim)
```

Comment:
I realized xlim is a better way of plotting a subset of the signal than using using plt.plot() with an index in the signal (i.e. x[idx_ts])). This way you can navigate within the plot manually to see the rest of the signal in case you want so.